pkg/dockerregistry/server/configuration/configuration.go

```diff
@@ -0,0 +1,105 @@
+package configuration
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/docker/distribution/configuration"
+)
+
+type openshiftConfig struct {
+	Openshift Configuration
+}
+
+type Configuration struct {
+	Version configuration.Version `yaml:"version"`
+	Metrics Metrics               `yaml:"metrics"`
+}
+
+type Metrics struct {
+	Enabled bool   `yaml:"enabled"`
+	Secret  string `yaml:"secret"`
+}
+
+func Parse(rd io.Reader) (*configuration.Configuration, *Configuration, error) {
+	in, err := ioutil.ReadAll(rd)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	openshiftEnv, err := popEnv("REGISTRY_OPENSHIFT_")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	dockerConfig, err := configuration.Parse(bytes.NewBuffer(in))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	dockerEnv, err := popEnv("REGISTRY_")
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := pushEnv(openshiftEnv); err != nil {
+		return nil, nil, err
+	}
+
+	config := openshiftConfig{}
+
+	p := configuration.NewParser("registry", []configuration.VersionedParseInfo{
+		{
+			Version: configuration.CurrentVersion,
```

Comment:
@miminar No. This is required to parse config file. Without it you will get [Unsupport version](https://github.com/docker/distribution/blob/master/configuration/parser.go#L125) error. Here, the entire upstream configuration is not important to us at all because we want to parse our own part of config file.